### ISSUE TITLE
facility decls w/ name matching enclosing-module-name causes codegen problems...

### ISSUE THREAD

**Issue by Welchd1** [Welchd1/resolve-lite#32]
Here's an interesting problem I stumbled into today. Look at this:

```
Facility Priorit_Fac;
        uses Standard_Integers, Standard_Booleans, Basic_Integer_Theory;
    Definition LEQV(x, y : Z) : B is x <= y;

    Operation Are_Ordered (restores i, j : Std_Integer_Fac :: Integer) :
            Std_Boolean_Fac :: Boolean;
        Procedure
            Are_Ordered := (i<=j);
    end Are_Ordered;

    Facility Priorit_Fac is Prioritizer_Template
            <Std_Integer_Fac :: Integer>(4, LEQV)
        implemented by Priorit_Heap_Impl(Are_Ordered);
...
```

attempting to generate code for this produces the following error:
```
cannot find symbol   symbol:   method Are_Ordered(RType,RType)   location: variable Priorit_Fac of type Prioritizer_Template
```

Generated code for the facility decl looks like this:

```
    public static final Prioritizer_Template Priorit_Fac = new Priorit_Heap_Impl(((Integer_Template)Standard_Integers.INSTANCE).initInteger(),
     ((Integer_Template)Standard_Integers.INSTANCE).initInteger(4),
     new BaseOperationParameter() {
            @Override public RType op(RType... e) {
                                return Priorit_Fac.Are_Ordered(e[0], e[1]);
            }
        });
```

the problem occurs at `return Priorit_Fac.Are_Ordered(e[0], e[1]);`. Confused? Well, turns out the name of the enclosing module is *also* `Priorit_Fac` (imagine that..). Anyways, we might need to restrict this. Not sure yet if something should change in codegen or the with semantic analysis..

**Comment by Welchd1** [comment]
Here's a potentially simple solution for this:

Add a 'dummy' symbol to module scope with the same name as the module. This way when someone tries to declare an operation, facility, etc matching the name of the module, it will trip a duplicate symbol exception...

This is all well and good, but I have concerns about the quality of the outgoing error message from this. It would be nice to simply special case the logic somewhere reliable: that is, somewhere where we can print a special duplicate symbol exception message informing people that they shouldn't name their things the same as the outmost module (really this is for codegen reasons primarily but I'm sure there's other places within actual resolve sourcecode that could create issues)..

**Comment by Welchd1** [comment]
Fixed (I think) as of cadf2fd4e9f89aea5ffe510b31e3c4e5a067f0b6. Reopen if needed.
